sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_receiver.py

```diff
@@ -640,10 +644,11 @@ def receive_messages(
         :param Optional[int] max_message_count: Maximum number of messages in the batch. Actual number
          returned will depend on prefetch_count and incoming stream rate.
          Setting to None will fully depend on the prefetch config. The default value is 1.
-        :param Optional[float] max_wait_time: Maximum time to wait in seconds for the first message to arrive.
-         If no messages arrive, and no timeout is specified, this call will not return
-         until the connection is closed. If specified, an no messages arrive within the
-         timeout period, an empty list will be returned.
+        :param Optional[float] max_wait_time: DEPRECATED. It is not advised to use this parameter.
```

Comment:
I think you could say something like, "It is not advised to use this parameter when calling this method. If you'd like to specify max_wait_time, please pass it in during client construction." To make it extra clear that the deprecation of this param is only happening in `receive_messages`, not in general.

Reply:
DEPRECATED. It is not advised to use this parameter when calling this method.  If you'd like to specify max_wait_time, please pass it in during client construction. If specified, this parameter interacts with the absolute operation timeout and will impact the amount of time alloted to retry the receive operation.

Reply:
wondering if we want to add detail on how it "impacts the amount of time"?

Reply:
maybe we could add a section elsewhere like to an issue and link in the changelog or in the docstring (if that is allowed) -- but to add it all here would make the docstring I think too confusing

Reply:
Ive been debating if we should add in a sample or a point to a test that would show how this change will impact someone using it.

Reply:
good point we should remove max_wait_time from recieve_messages in all the samples and then make a sample/test that runs receive_messages and throws an error and show how it only retries like 1 or 2 times if the max_wait_time is set to be small